python/ray/tune/experiment/trial.py

```diff
@@ -749,23 +749,46 @@ def pickled_error_file(self):
             return None
         return os.path.join(self.local_path, self.run_metadata.pickled_error_filename)
 
-    def handle_error(self, exc: Optional[Union[TuneError, RayTaskError]] = None):
+    def _handle_restore_error(self, exc: _TuneRestoreError):
+        exc = exc.exc
+        if self.temporary_state.num_restore_failures >= int(
+            os.environ.get("TUNE_RESTORE_RETRY_NUM", 0)
+        ):
+            # Restore was unsuccessful, try again without checkpoint.
+            self.clear_checkpoint()
+            self.run_metadata.num_failures += 1
+        else:
+            self.temporary_state.num_restore_failures += 1
+
+    def _handle_ray_actor_error(self, exc: RayActorError):
+        exc._preempted = True  # TODO(justinvyu): Test the real integration
+        if not exc._preempted:
+            # Only count non-preempted actor errors as failures.
+            self.run_metadata.num_failures += 1
+
+    def _handle_ray_task_error(self, exc: RayTaskError):
+        if isinstance(exc.cause, RayActorError):
+            # Handle the RayActorError directly (ex: Ray Train worker actor errors)
+            return self._handle_ray_actor_error(exc.cause)
+
+        # Increment failures for all user errors (which get raised as RayTaskError)
+        self.run_metadata.num_failures += 1
+
+    def handle_error(
+        self, exc: Optional[Union[TuneError, RayTaskError, RayActorError]] = None
+    ):
         if isinstance(exc, _TuneRestoreError):
-            exc = exc.exc
-            if self.temporary_state.num_restore_failures >= int(
-                os.environ.get("TUNE_RESTORE_RETRY_NUM", 0)
-            ):
-                # Restore was unsuccessful, try again without checkpoint.
-                self.clear_checkpoint()
-                self.run_metadata.num_failures += 1
-            else:
-                self.temporary_state.num_restore_failures += 1
+            self._handle_restore_error(exc)
+        elif isinstance(exc, RayActorError):
+            self._handle_ray_actor_error(exc)
+        elif isinstance(exc, RayTaskError):
+            self._handle_ray_task_error(exc)
         else:
             self.run_metadata.num_failures += 1
 
         if self.local_path:
             self.run_metadata.error_filename = EXPR_ERROR_FILE
-            if isinstance(exc, RayTaskError):
+            if isinstance(exc, (RayTaskError, RayActorError)):
```

Comment:
Hmmm given that we've never logged these before, when does `RayActorError` actually happen? Would it be a `RayActorError` or `RayTaskError` if the trial node gets preempted?

Reply:
If the trial node gets preempted, it's a `RayActorError`.

```
ray.get(A.task.remote()) -> RayActorError if A's node dies
ray.get(A.task.remote()) -> RayTaskError(OriginalError) if A.task raises an OriginalError inside it.
```

Reply:
I think it was just an oversight not to log `RayActorError`